### PR TITLE
INTMDB-419-network peering

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

INTMDB-419 network peering upgrade

- Added Validations for all method inputs. 
- Added error handling.

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 1]
```diff
@@ -117,9 +124,28 @@ func validateOrCreateNetworkContainer(req *handler.Request, prevModel *Model, cu
...
+var ReadRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "ProjectId", "Id"}
+var UpdateRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "ProjectId", "AccepterRegionName",
+	"AwsAccountId", "RouteTableCIDRBlock"}
+var DeleteRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "ProjectId", "Id"}
+var ListRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "ProjectId"}
+
+// function to validate inputs to all actions
+func validateModel(fields []string, model *Model) *handler.ProgressEvent {
+	return validator.ValidateModel(fields, model)
+}
```
Comment (andreaangiolillo) [inline cfn-resources/network-peering/cmd/resource/resource.go:138]: Please define


```
const (
	publicKey    = "ApiKeys.PublicKey"
	privateKey   = "ApiKeys.PrivateKey"
	projectId    = "ProjectId"
......
)
```
then 
```
var CreateRequiredFields = []string{publicKey, privateKey, projected, .....}
```

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 2]
```diff
@@ -117,9 +124,28 @@ func validateOrCreateNetworkContainer(req *handler.Request, prevModel *Model, cu
...
+
 // Create handles the Create event from the Cloudformation service.
 func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
 	setup()
+
+	modelValidation := validateModel(CreateRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
```
Comment (andreaangiolillo) [inline cfn-resources/network-peering/cmd/resource/resource.go:147]:
```suggestion
	if modelValidation := validateModel(CreateRequiredFields, currentModel); modelValidation != nil {
		return *modelValidation, nil
	}
```

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 3]
```diff
@@ -191,6 +214,11 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
 // Read handles the Read event from the Cloudformation service.
 func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
 	setup()
+	modelValidation := validateModel(ReadRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
```
Comment (andreaangiolillo) [inline cfn-resources/network-peering/cmd/resource/resource.go:220]:
```suggestion
	if modelValidation := validateModel(CreateRequiredFields, currentModel); modelValidation != nil {
		return *modelValidation, nil
	}
```

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 4]
```diff
@@ -254,6 +271,11 @@ func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
 func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
 	setup()
 	log.Debugf("Update currentModel:%+v", currentModel)
+
+	modelValidation := validateModel(UpdateRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
```
Comment (andreaangiolillo) [inline cfn-resources/network-peering/cmd/resource/resource.go:278]:
```suggestion
	if modelValidation := validateModel(CreateRequiredFields, currentModel); modelValidation != nil {
		return *modelValidation, nil
	}
```

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 5]
```diff
@@ -322,6 +333,11 @@ func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler
 // Delete handles the Delete event from the Cloudformation service.
 func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
 	setup()
+	modelValidation := validateModel(DeleteRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
```
Comment (andreaangiolillo) [inline cfn-resources/network-peering/cmd/resource/resource.go:339]:
```suggestion
	if modelValidation := validateModel(CreateRequiredFields, currentModel); modelValidation != nil {
		return *modelValidation, nil
	}
```

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 6]
```diff
@@ -368,6 +373,10 @@ func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler
 // List handles the List event from the Cloudformation service.
 func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
 	setup()
+	modelValidation := validateModel(ListRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
```
Comment (andreaangiolillo) [inline cfn-resources/network-peering/cmd/resource/resource.go:379]:
```suggestion
	if modelValidation := validateModel(CreateRequiredFields, currentModel); modelValidation != nil {
		return *modelValidation, nil
	}
```

File: cfn-resources/util/validator/validator_test.go [hunk 7]
```diff
@@ -1,11 +1,16 @@
...
-	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
 	"testing"
 )
 
+var CreateRequiredFields = []string{""}
+var ReadRequiredFields = []string{"ProjectId", "ClusterName", "Id", "ApiKeys.PublicKey", "ApiKeys.PrivateKey"}
+var UpdateRequiredFields []string
+var DeleteRequiredFields = []string{"ProjectId", "ClusterName", "Id", "ApiKeys.PublicKey", "ApiKeys.PrivateKey"}
+var ListRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "ClusterName", "ProjectId"}
```
Comment (andreaangiolillo) [inline cfn-resources/util/validator/validator_test.go:12]: Same as above

File: cfn-resources/network-peering/cmd/resource/resource.go [hunk 8]
```diff
@@ -453,3 +351,91 @@ func networkPeeringIsReady(client *mongodbatlas.Client, projectID, peerID, targe
...
+	containerRequest.RegionName = *region
+	containerRequest.ProviderName = "AWS"
+	containerRequest.AtlasCIDRBlock = DefaultAWSCIDR
+	_, _ = logger.Debugf("containerRequest:%+v", containerRequest)
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		return container, err
+	}
+	containerResponse, resp, err := client.Containers.Create(context.TODO(), *currentModel.ProjectId, containerRequest)
+	// TODO add logging here
```
Comment (andreaangiolillo) [inline cfn-resources/network-peering/cmd/resource/resource.go:426]:
```suggestion
```

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Review by Zuhairahmed — CHANGES_REQUESTED** [review]
Missing updates to docs + README file, can you please add to this PR?

**Review by Zuhairahmed — APPROVED** [review]
LGTM

**Review by ParthasarathyVaradhan — APPROVED** [review]
LGTM

**Pull request merged**